pwndbg/gdblib/disasm/arm.py

```diff
@@ -61,16 +64,42 @@
     ARM_INS_ORR: "|",
     ARM_INS_AND: "&",
     ARM_INS_EOR: "^",
-    ARM_INS_ASR: ">>s",
-    ARM_INS_LSR: ">>",
-    ARM_INS_LSL: "<<",
     ARM_INS_UDIV: "/",
     ARM_INS_SDIV: "/",
     ARM_INS_MUL: "*",
     ARM_INS_UMULL: "*",
     ARM_INS_SMULL: "*",
 }
 
+ARM_SHIFT_INSTRUCTIONS = {
+    ARM_INS_ASR: ">>s",
+    ARM_INS_LSR: ">>",
+    ARM_INS_LSL: "<<",
+}
+
+
+def first_op_is_pc(i: PwndbgInstruction) -> bool:
+    return i.operands[0].reg == ARM_REG_PC
+
+
+def ops_contain_pc(i: PwndbgInstruction) -> bool:
+    for op in i.operands:
+        if op.type == CS_OP_REG:
+            if op.reg == ARM_REG_PC:
+                return True
+    return False
+
+
+ARM_CAN_WRITE_TO_PC: Dict[int, Callable[[PwndbgInstruction], bool]] = {
+    ARM_INS_ADD: first_op_is_pc,
+    ARM_INS_SUB: first_op_is_pc,
+    ARM_INS_SUBS: first_op_is_pc,
+    ARM_INS_MOV: first_op_is_pc,
+    ARM_INS_LDR: first_op_is_pc,
+    ARM_INS_POP: ops_contain_pc,
+    ARM_INS_LDM: ops_contain_pc,
```

Comment:
Random thought: we could document instructions here somehow for easier understanding for future readers or link them? Idk, just loud thinking; we don't necessarily have to do it